src/NuGet.CommandLine.StagingExtensions/StageCommand.cs

```diff
@@ -46,16 +60,53 @@ public class StageCommand : Command
 
             var stageManagementResource = await GetStageManagementResource(source);
 
-            if (!string.IsNullOrEmpty(Create))
+            if (subCommand == SubCommand.Create)
             {
-                await CreateStage(stageManagementResource, apiKey, Create);
+                await CreateStage(stageManagementResource, apiKey, parameter);
             }
-            else if (!string.IsNullOrEmpty(Drop))
+            else if (subCommand == SubCommand.Drop)
             {
-                await DropStage(stageManagementResource, apiKey, Drop);
+                await DropStage(stageManagementResource, apiKey, parameter);
+            }
+            else if (subCommand == SubCommand.List)
+            {
+                await ListStages(stageManagementResource, apiKey);
             }
         }
 
+        private bool TryParseArguments(out SubCommand subCommand, out string parameter)
+        {
+            subCommand = SubCommand.None;
+            parameter = string.Empty;
+
+            string commandString = Arguments[0];
+
+            if (string.Equals(commandString, "create", StringComparison.InvariantCultureIgnoreCase))

```

Comment:
You could use [`Enum.Parse`](https://msdn.microsoft.com/en-us/library/kxydatf9(v=vs.110).aspx) to do the work of this and the following `if` statements.